taichi/transforms/simplify.cpp

```diff
@@ -1162,7 +1162,6 @@ void full_simplify(IRNode *root, const CompileConfig &config, Kernel *kernel) {
     alg_simp(root, config);
     die(root);
     whole_kernel_cse(root);
-    variable_optimization(root);
```

Comment:
Interestingly, avoid doing some variable optimizations leads to final optimization on some tests.

Reply:
Right, now you need something like https://ieeexplore.ieee.org/document/1611550